### PR TITLE
Rename analysis result to analysis summary

### PR DESCRIPTION
**Pull request by charisk**
Currently the term "analysis result" is used to refer to the summary of an analysis outcome. However, the term "result" means something else in terms of an analysis - it means an actual result of the query. 

I've renamed analysis result to analysis summary in order to avoid confusion.

## Checklist
N/A 
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] If this pull request makes user-facing changes that require documentation changes, the `ready-for-doc-review` label has been added to this pull request or the corresponding issue.

### CODE REVIEW

File: extensions/ql-vscode/src/remote-queries/remote-queries-interface.ts [hunk 1]
```diff
@@ -259,11 +259,11 @@ export class RemoteQueriesInterfaceManager {
   }
 
   /**
-   * Builds up a list of analysis results, in a data structure tailored to the view.
-   * @param analysisResults The results of a specific analysis.
+   * Builds up a list of analysis summaries, in a data structure tailored to the view.
+   * @param analysisResults The summary of a specific analysis.
```
Comment (shati-patel) [inline extensions/ql-vscode/src/remote-queries/remote-queries-interface.ts:263]: Very minor: Would it make sense for this param to be called `analysisSummaries` too? 🔍

File: extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts [hunk 2]
```diff
@@ -92,7 +92,7 @@ export class RemoteQueriesManager {
 
     return {
       executionEndTime,
-      analysisResults,
+      analysisSummaries: analysisResults,
```
Comment (shati-patel) [inline extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts:95]: As above, maybe `analysisResults` -> `analysisSummaries` would be more consistent?

**Review by shati-patel — APPROVED** [review]
Our terminology is a bit confusing 🙃  Thanks for renaming! 🧹 

(Left another tiny potential renaming suggestion, but feel free to merge as-is ⚡ )

**Comment by charisk** [comment]
Thanks for the speedy review @shati-patel! I've tided up the remaining bits.

**Review by shati-patel — APPROVED** [review]
Thanks! 😸

**Pull request merged**